### PR TITLE
Added a loss cone population DRO for precipitation

### PR DESCRIPTION
**Pull request by maximegrandin**
Added a data reduction operator to save the loss-cone population of ion species in each cell.
- At the moment, the number of energy bins (16) as well as the minimum and maximum bin values (100 eV and 100 keV) are hardcoded. Energy bins are logarithmically spaced. Particles with energies under 100 eV (above 100 keV) are counted with the lowest (highest) energy bin.
- The loss-cone opening angle has a fixed value which is the same in the whole simulation domain. The value was set to 10 deg, which should be ok as a first approximation, since the phase-space density is averaged inside the loss cone (not integrated).
- At the moment, the loss-cone population is saved everywhere, including in regions where it is not relevant for particle precipitation.
- Differential ion number flux values are saved in units used by observers [ion/(cm2 s sr eV)], which is a deviation from the convention of manipulating SI units only.

### CODE REVIEW

**Comment by ykempf** [comment]
I suggest that you make the hard-coded parameters cfg parameters. It's best if someone on-site shows you the basics, or if you have a bit of time, look at how it's done in parameters.cpp and parameters.h. Add a declaration (add) and a retrieval function (get) like other such scalar parameters, as part of the variables group.

File: testpackage/tests/Magnetosphere_polar_small/Magnetosphere_polar_small.cfg [hunk 1]
```diff
@@ -32,7 +32,7 @@ y_min = -5.0e6
 y_max = 5.0e6
 z_min = -250.0e6
 z_max = 250.0e6
-t_max = 20.05
+t_max = 1000.05
```
Comment (markusbattarbee) [inline testpackage/tests/Magnetosphere_polar_small/Magnetosphere_polar_small.cfg]: I think we don't want to edit the testpackage cfg files ;)

File: datareduction/datareductionoperator.h [hunk 2]
```diff
@@ -507,6 +507,25 @@ namespace DRO {
...
+      virtual bool reduceData(const SpatialCell* cell,char* buffer);
+      virtual bool setSpatialCell(const SpatialCell* cell);
+      
+   protected:
+      uint popID;
+      std::string popName;
+      int nChannels;
+      Real emin, emax;
+      Real cosAngle;
+      std::vector<Real> channels, dataDiffFlux;
```
Comment (markusbattarbee) [inline datareduction/datareductionoperator.h]: You don't actually end up using the channels vector for anything

**Comment by markusbattarbee** [comment]
Note, parameters.cpp needs to also be updated with the name of the data reducer

**Comment by ykempf** [comment]
> Note, parameters.cpp needs to also be updated with the name of the data reducer

What do you mean @markusbattarbee?

**Comment by markusbattarbee** [comment]
See new pull request introducing how DROs can write parameters to the .vlsv file

**Comment by maximegrandin** [comment]
Will be included in upcoming pull request.

**Pull request closed without merging**